### ISSUE TITLE
Selections broke after deleting text using vscode.TextEditor.edit

### ISSUE THREAD

**Issue by aioutecism** [Microsoft/vscode#15236]
- VSCode Version: 1.8.0-insider
- OS Version: OSX 10.11.6 (15G1108)

Steps to Reproduce:

1. Create an extension.
2. Implement a command like this:
```
// package.json
{
    // ...
    "contributes": {
        "commands": [
            {
                "command": "customDelete",
                "title": "Custom Delete"
            }
        ]
    }    
    // ...
}
```
```ts
// extension.ts
import {commands, window, ExtensionContext} from 'vscode';

export function activate(context: ExtensionContext) {
    commands.registerCommand('customDelete', () => {
        const activeTextEditor = window.activeTextEditor;

        activeTextEditor.edit(editBuilder => {
            activeTextEditor.selections.forEach(selection => {
               editBuilder.delete(selection);
            });
        });
    });
}
```
3. Debug the extension.
4. Enter some text:
```
foofoofoo, foofoofoo, bar
```
5. Select those two `foofoofoo` using `cmd+d`.
6. Run the implemented command.
7. Selections are broken.

Screen record:
![2016-11-09 18 55 43](https://cloud.githubusercontent.com/assets/2175401/20134274/34c68862-a6ae-11e6-8f81-bad3a74d4d82.gif)

**Comment by aioutecism** [comment]
```ts
editBuilder.replace(selection, '');
```
Replace has no luck too.
However, `commands.executeCommand('deleteRight');` works correctly.

**Comment by aioutecism** [comment]
Fixed in the latest Insider build.
(Though don't know which commit does the fix.)

**Comment by alexandrudima** [comment]
@aioutecism This was fixed by dc961cbc4e1a005ee3073b61c5c566097d372d82, nevertheless I've just pushed a new test to also cover this case

**Comment by aioutecism** [comment]
@alexandrudima Thanks for the information.
